test/parallel/test-fs-buffer.js

```diff
@@ -29,13 +29,17 @@ assert.throws(() => {
 }, /path must be a string or Buffer/);
 
 const dir = Buffer.from(common.fixturesDir);
-fs.readdir(dir, 'hex', common.mustCall((err, list) => {
+fs.readdir(dir, 'hex', common.mustCall((err, hexList) => {
   assert.ifError(err);
-  list = list.map((i) => {
-    return Buffer.from(i, 'hex').toString();
-  });
-  fs.readdir(dir, common.mustCall((err, list2) => {
+  fs.readdir(dir, common.mustCall((err, stringList) => {
     assert.ifError(err);
-    assert.deepStrictEqual(list, list2);
+    stringList.forEach((val, idx) => {
+      const fromHexList = Buffer.from(hexList[idx], 'hex').toString();
+      assert.strictEqual(
+        fromHexList,
+        val,
+        `${hexList[idx]} is hex value for ${fromHexList} and not ${val}`
```

Comment:
How about this?:

```js
`expected ${val}, got ${fromHexList} by hex decoding ${hexList[idx]}`
```

The thing I'm trying to convey is this: ${val} is the input. We put it in a file hex encoded. We read the file back in and got the hex-encoded value ${hexList[idx]}. When we hex decode that value, we got {fromHexList} instead of ${val}.